csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -52,6 +55,131 @@ public Builder DataType(IArrowType type)
                 return this;
             }
 
+            // Build DataType from C# Type
+            public Builder DataType(System.Type valueType, string timezone = null)
```

Comment:
The `timezone` parameter is a little odd, especially since it gets passed down into nested types.

Reply:
added default timezone, and dict for specific fields